### ISSUE TITLE
Support for >=, <=, >, < conditionals. 

### ISSUE THREAD

**Issue by vicaire** [argoproj/argo#860]
**Is this a BUG REPORT or FEATURE REQUEST?**: FEATURE REQUEST

Argo conditionals seem to only support "==": 

https://github.com/argoproj/argo/blob/master/examples/conditionals.yaml

Would it make sense to add support for additional operations such as >=, <=, >, <?

For instance, in a machine learning context: 

- One step computes a model
- The model is stored in an object store
- the step outputs the accuracy of the model. 
- We could then use the ">=" to express that the workflow should proceed to the next step only if the accuracy of the model is greater than X.

**Comment by vicaire** [comment]
@qimingj

**Comment by jessesuen** [comment]
> Would it make sense to add support for additional operations such as >=, <=, >, <?

Yes. The conditional parsing in Argo is extremely dumb at the moment.  To go even further, we should support grouping with parens. Kubernetes already provides a very nice parsing library for this: 
`"k8s.io/apimachinery/pkg/labels"`

This library is already being used in Argo's resource templates to support success conditions and failure conditions, but has not yet made its way in the controller conditional evaluation logic.

**Comment by vicaire** [comment]
Grouping with parens? Would that be to support multiple output parameters?

**Comment by jessesuen** [comment]
Right. You should be able to do something like:
(foo == true) && (bar == true)

**Comment by jessesuen** [comment]
Implemented in 9b5c85631765285b4593b7707ede014178f77679 using the library: https://github.com/Knetic/govaluate. Here are some examples of what is now supported:

```
		"foo == foo",
		"foo != bar",
		"1 == 1",
		"1 != 2",
		"1 < 2",
		"1 <= 1",
		"a < b",
		"(foo == bar) || (foo == foo)",
		"(1 > 0) && (1 < 2)",
		"Error in (Failed, Error)",
		"!(Succeeded in (Failed, Error))",
		"true == true",
```

**Comment by vicaire** [comment]
Thanks jessesuen!

**Comment by sylock** [comment]
The implemented functionality is really nice but the documentation should be upgraded to explain it. Currently the documentation is very poor about it. You only have :
* a reference in the beginner core concept: https://argoproj.github.io/argo-workflows/workflow-concepts/#template-invocators
* one example that does not explain all the possibilities: https://argoproj.github.io/argo-workflows/examples/#conditionals

In conclusion: this current issue is the only place where you can find a good documentation about the conditionals syntax.

**Comment by simster7** [comment]
@sylock Thanks! Would you like to submit a PR with better documentation about this?

**Comment by julien-blanchon** [comment]
I didn't manage to make this work using :+1: 
```yaml
when: {{steps.predict.outputs.artifacts.score}}"
```
I get the following workflow error:
```
Invalid 'when' expression '{{steps.predict.outputs.artifacts.score}} > 0.5': Invalid token: '{{' (hint: try wrapping the affected expression in quotes ("))
```
